### PR TITLE
avoid option.acquireTimeout when using mysql2

### PR DESCRIPTION
**Pull request by rchaser53**
related: https://github.com/typeorm/typeorm/issues/3127

### CODE REVIEW

**Comment by rchaser53** [comment]
I have started using this library today. But this library emits the warning message when using with mysql2. It's a bit noisy. So I sent the this PR.

File: src/driver/mysql/MysqlDriver.ts [hunk 1]
```diff
@@ -802,7 +808,7 @@ export class MysqlDriver implements Driver {
         } catch (e) {
             try {
                 this.mysql = PlatformTools.load("mysql2"); // try to load second supported package
-
+                this.version = 2;
```
Comment (johannesschobel) [inline src/driver/mysql/MysqlDriver.ts]: can we have the same order as above (first set version, then load packages) or vice-versa.. but not mixed?

File: src/driver/mysql/MysqlDriver.ts [hunk 2]
```diff
@@ -829,7 +834,12 @@ export class MysqlDriver implements Driver {
             debug: options.debug,
             trace: options.trace,
             multipleStatements: options.multipleStatements,
-            flags: options.flags
+            flags: options.flags,
+            ...(
+                this.version === 1
+                    ? { acquireTimeout: options.acquireTimeout } 
+                    : {}
+                )
```
Comment (johannesschobel) [inline src/driver/mysql/MysqlDriver.ts]: i think this statement is particularly hard to read.. but that is just my opinion

File: src/driver/mysql/MysqlDriver.ts [hunk 3]
```diff
@@ -788,6 +793,7 @@ export class MysqlDriver implements Driver {
      */
     protected loadDependencies(): void {
         try {
+            this.version = 1;
```
Comment (johannesschobel) [inline src/driver/mysql/MysqlDriver.ts]: i would rather not use `1`, `2` as version tags but rather `mysql`, `mysql2` and so on..
further, can we use this maybe as an enum for checks later on?

**Comment by rchaser53** [comment]
Thank you the review. I fixed it.

File: src/driver/mysql/MysqlDriver.ts [hunk 4]
```diff
@@ -37,6 +45,11 @@ export class MysqlDriver implements Driver {
      */
     mysql: any;
 
+    /**
+     * Mysql version.
+     */
+    version: MySQLVersion;
```
Comment (pleerock) [inline src/driver/mysql/MysqlDriver.ts]: I'm sorry to annoy you but we don't really need enum here. Regular string literal type is enough. (e.g. replace `version: MySQLVersion` to `version: "mysql"|"mysql2")

**Comment by pleerock** [comment]
Thank you for your contribution! I have left one comment, but its not critical to fix. If you have time to fix that please send another PR while I'm going to merge this one.

**Comment by rchaser53** [comment]
Good idea! I changed my PR.

**Comment by pleerock** [comment]
Actually I meant just traverse over all object properties and filter out undefined onces. But thats fine too.

**Comment by pleerock** [comment]
Thank you!

**Pull request merged**